pkg/tagger/collectors/workloadmeta_extract.go

```diff
@@ -102,82 +105,94 @@ var (
 )
 
 func (c *WorkloadMetaCollector) processEvents(evBundle workloadmeta.EventBundle) {
-	var tagInfos []*TagInfo
+	tagInfos := make([]*TagInfo, 0, len(evBundle.Events))
 
 	for _, ev := range evBundle.Events {
 		entity := ev.Entity
 		entityID := entity.GetID()
 
+		var tagInfo *TagInfo
+
 		switch ev.Type {
 		case workloadmeta.EventTypeSet:
-			taggerEntityID := buildTaggerEntityID(entityID)
-
-			// keep track of children of this entity from previous
-			// iterations ...
-			unseen := make(map[string]struct{})
-			for childTaggerID := range c.children[taggerEntityID] {
-				unseen[childTaggerID] = struct{}{}
-			}
-
-			// ... and create a new empty map to store the children
-			// seen in this iteration.
-			c.children[taggerEntityID] = make(map[string]struct{})
-
 			switch entityID.Kind {
 			case workloadmeta.KindContainer:
-				tagInfos = append(tagInfos, c.handleContainer(ev)...)
+				tagInfo = c.handleContainer(ev)
 			case workloadmeta.KindKubernetesPod:
-				tagInfos = append(tagInfos, c.handleKubePod(ev)...)
+				tagInfo = c.handleKubePod(ev)
 			case workloadmeta.KindECSTask:
-				tagInfos = append(tagInfos, c.handleECSTask(ev)...)
+				tagInfo = c.handleECSTask(ev)
 			case workloadmeta.KindGardenContainer:
-				tagInfos = append(tagInfos, c.handleGardenContainer(ev)...)
+				tagInfo = c.handleGardenContainer(ev)
 			default:
 				log.Errorf("cannot handle event for entity %q with kind %q", entityID.ID, entityID.Kind)
 			}
 
-			// remove the children seen in this iteration from the
-			// unseen list ...
-			for childTaggerID := range c.children[taggerEntityID] {
-				delete(unseen, childTaggerID)
-			}
-
-			// ... and remove entities for everything that has been
-			// left
-			source := buildTaggerSource(entityID)
-			tagInfos = append(tagInfos, c.handleDeleteChildren(source, unseen)...)
-
 		case workloadmeta.EventTypeUnset:
-			tagInfos = append(tagInfos, c.handleDelete(ev)...)
+			tagInfo = c.handleDelete(ev)
 
 		default:
 			log.Errorf("cannot handle event of type %d", ev.Type)
 		}
 
+		if tagInfo != nil {
+			tagInfos = append(tagInfos, tagInfo)
+		}
 	}
 
-	// NOTE: haha, this is still async and race conditions will still
-	// happen :D since the workloadmeta will be the only collector in the
-	// tagger in the end, this can be turned into a sync call to
-	// processTagInfo
 	if len(tagInfos) > 0 {
-		c.out <- tagInfos
+		c.p.ProcessTagInfo(tagInfos)
```

Comment:
with `c.p` is not clear what is `p`, can we rename it with a more self explain name?